### DISCUSSION TITLE
Set filter for table with a record's column action

### DISCUSSION THREAD

**Discussion by robinheidrich** [filamentphp/filament#13273] [Help]
### Package

Table builder

### Package Version

v3.2.91

### How can we help you?

Is it possible to set the filter for a table using the action of a record's column? I can't find anything about this in the documentation.

Let's say I have a filter for `user_id`, when I click on the column I want this filter to be set to the `user_id` of the record.

**Comment by leandrocfe** [comment]
if you have a filter like this

```php
->filters([
    Tables\Filters\SelectFilter::make('user_id')
        ->relationship('user', 'name'),

])
```

you could add a `url()` in the column

```php
->url(fn (YourModel $record) => YourResource::getUrl('index', ['tableFilters' => ['user_id' => ['value' => $record->user_id]]]))
```

**Reply by robinheidrich** [reply]
I hadn't even thought of that! Works perfectly, but:

1. It doesn't show that filters are active, only on the filter icon. I guess this hasn't been implemented (yet)?
2. How do I set multiple filters? Tried everything and couldn't get it to work.
3. Other filters should stay in place. This would remove all other filters.
